File: 234-palindrome-linked-list/palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        prev=head
        list_l = []

        while prev:
            list_l.append(prev.val)
            prev= prev.next
        

        i=0
        l=len(list_l)-1

        while i<l:
            if list_l[i] != list_l[l]:
                return False

            else :
                i+=1
                l-=1
        return True

File: 234-palindrome-linked-list/palindrome-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
        ListNode curr = head;
        ListNode ncurr = head;

        Stack<Integer> stk = new Stack<Integer>();

        while(curr != null){
            stk.push(curr.val);
            curr = curr.next;
        }

        while(ncurr!=null){
            if(stk.pop() != ncurr.val){
                return false;
            }
            ncurr=ncurr.next;
        }

        return true;
    }
}